OpenRA.Game/Primitives/RingBuffer.cs

```diff
@@ -0,0 +1,145 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenRA.Primitives
+{
+	/// <summary>Fixed size rorating buffer backed by an array.</summary>
+	public class RingBuffer<T> : ICollection<T>, IEnumerable<T>
+	{
+		readonly IComparer<T> comparer;
+		readonly T[] values;
+		int start;
+
+		public int Capacity => values.Length;
+		public int Count { get; private set; }
+		public bool IsReadOnly => false;
+
+		public RingBuffer(int capacity, IComparer<T> comparer)
+		{
+			this.comparer = comparer;
+			#pragma warning disable IDE0090
```

Comment:
Why is this necessary?

Reply:
couldn't find what the `new()` syntax for arrays. `new ()[size]` seemed wrong?

Reply:
My IDE claims the suppression is not needed:
![image](https://github.com/OpenRA/OpenRA/assets/7704140/84b7d0c5-8597-41c1-99fc-63d0936e4b1b)
Seems like you already fixed the issue somewhen after adding the pragmas.

Reply:
I get no warning after removing the pragmas:
![image](https://github.com/OpenRA/OpenRA/assets/7704140/f25b9446-d70c-4209-827f-4bf02b5cd696)